### PR TITLE
feat: add the step configuration page, plus the required helpers 

### PR DESCRIPTION
**Pull request by riccardo-forina**
Also, some refactoring.

The `WithIntegrationHelpers` has been changed to simply expose the new hook [`useIntegrationHelpers`](https://github.com/syndesisio/syndesis-react/blob/f8eb0e79662cd8c20f214c153b3acb9c813ce2f3/app/ui-react/packages/api/src/useIntegrationHelpers.tsx).

Two new helpers have been added, `addStep` and `updateStep`. Since I was there, I copy over the integration functions from the Angular codebase and used some in the new helpers. Ideally, the old functions should be updated to make use of the helpers too, like for saving I see there are functions to ensure that the integration object is valid and update the tags, we should use those.

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1254/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1260/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/packages/api/src/helpers/integrationFunctions.ts [hunk 1]
```diff
@@ -200,6 +193,454 @@ export function getStepIcon(apiUri: string, step: Step): string {
...
+    outputDataShape.kind.toLowerCase() === DataShapeKinds.ANY
+  );
+}
+
+/**
+ * Filters connections based on the supplied position in the step array
+ * @param steps
+ * @param position
+ */
+export function filterStepsByPosition(
```
Comment (gashcrumb) [inline app/ui-react/packages/api/src/helpers/integrationFunctions.ts]: We actually already have this imported here -> https://github.com/syndesisio/syndesis-react/blob/master/app/ui-react/syndesis/src/modules/integrations/components/editor/utils.ts#L251 and it's being used already.

Comment (gashcrumb) [reply app/ui-react/packages/api/src/helpers/integrationFunctions.ts]: but it's cool, I'll pull this one out next PR :-)

**Comment by riccardo-forina** [comment]
Btw, this closes #250

**Pull request merged**